src/Client/Grpc/GrpcDurableTaskClient.cs

```diff
@@ -86,14 +86,22 @@ public override async Task<string> ScheduleNewOrchestrationInstanceAsync(
             Input = this.DataConverter.Serialize(input),
         };
 
-        if (Activity.Current?.Id != null)
+        if (Activity.Current?.Id != null || Activity.Current?.TraceStateString != null)
         {
-            request.ParentTraceContext.TraceParent = Activity.Current?.Id;
-        }
+            if (request.ParentTraceContext == null)
```

Comment:
Might be able to consolidate this to:

``` CSharp
if (Activity.Current is { } activity)
 {
    request.ParentTraceContext = new()
    {
        TraceParent = activity.Id,
        TraceState = activity.TraceStateString,
    };
}
```

Reasons being:

1. We don't care if `ParentTraceContext` is already set or not, we can overwrite it.
2. if `Activity` is not null, we don't care if Id or TraceStateString are null or not - just set them.